merge_sort.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+import time
+import random
+
+
+def timed_func(func):
+    """Decorator for timing our traversal methods."""
+    def timed(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        elapsed = time.time() - start
+        print "time expired: %s" % elapsed
+        return result
+    return timed
+
+
+@timed_func
+def time_merge_sort(sequence):
+    return merge_sort(sequence)
+
+
+def merge_sort(sequence):
+    """Mergesort."""
+    for ind, num in enumerate(sequence):
+        # base case
+        if len(sequence) <= 1:
+            return sequence
+        # recursive case
+        middle = len(sequence) / 2
+        left = merge_sort(sequence[:middle])
+        right = merge_sort(sequence[middle:])
+
+        return merge(left, right)
+
+
+def merge(left, right):
+    result =[]
+    while left and right:
+        if left[0] <= right[0]:
+            result.append(left.pop(0))
+        else:
+            result.append(right.pop(0))
```

Comment:
The `list.pop(0)` operation above here is O(n) and will destroy any sort of performance boost you get from the mergesort algorithm.  It'd be great to come up with a way not to need to do that.